### ISSUE TITLE
Rule change: array-element-newline default option before multiline

### ISSUE THREAD

**Issue by Zamralik** [eslint/eslint#13818]
**What rule do you want to change?**

array-element-newline

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option "consistent".
Similar to `object-curly-newline`
```json
"array-element-newline": ["error", { "consistent": true, "multiline": true }]
```

if multiline applies, behavior is "always"
else if consistent is true, behavior is "consistent".
else behavior is "never".

**Please provide some example code that this change will affect:**

```js
const numbers = [1, 2, 3, 4];
const strings = [
    "Hello, World!",
    "Lorem ipsum dolor sit amet",
];
const matrix = [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
];
// The only one that should be invalid
const objects = [
    {
        foo: 0,
    }, {
        foo: 1,
    },
];
```

**What does the rule currently do for this code?**

With rule
```json
"array-element-newline": ["error", { "multiline": true }]
```
`strings` and `matrix` are invalid.

With rule
```json
"array-element-newline": ["error", "consistent"]
```
`objects` is valid.

**What will the rule do after it's changed?**

`numbers`, `strings`, and `matrix` would be valid, but `objects` would be invalid.

**Are you willing to submit a pull request to implement this change?**

No, I do not know how ESLint works internally.

**Comment by mdjermanovic** [comment]
Thanks for the issue!

Per our [rule policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), stylistic rules are frozen.

Nevertheless, I think we should consider making this change, especially because this rule doesn't take into account the length of the elements, which might be an oversight in the design (the original proposal #6075 had `minColumn` option).

I believe the `multiline: true` option would be more useful if it wouldn't strictly disallow linebreaks between elements when there are no multiline elements.

For example, this code is invalid by the actual rule:

```js
/*eslint array-element-newline: ["error", { "multiline": true }]*/

const strings = [
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  "sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
];
```

and should be fixed to a long line with all elements:

```js
const strings = [
  "Lorem ipsum dolor sit amet, consectetur adipiscing elit", "sed do eiusmod tempor incididunt ut labore et dolore magna aliqua",
];
```

**Comment by Zamralik** [comment]
I improved the example with a better formatting for the number[][] matrix.
